vm/src/obj/objsequence.rs

```diff
@@ -163,29 +163,25 @@ pub fn get_item(
     }
 
     if subscript.payload::<PySlice>().is_some() {
-        let payload = if sequence.payload::<PyList>().is_some() {
-            PyObjectPayload::AnyRustValue {
-                value: Box::new(PyList::from(
-                    elements.to_vec().get_slice_items(vm, &subscript)?,
-                )),
-            }
+        if sequence.payload::<PyList>().is_some() {
```

Comment:
We should use `isinstance` to check on types, not on payload. But we can fix this in another PR.